src/thread_task.rs

```diff
@@ -566,7 +623,7 @@ unsafe fn check_tile(
                     .fetch_or((p3 == FRAME_ERROR) as c_int, Ordering::SeqCst);
             }
             // next:
-            tasks[t_idx].deps_skip += 1;
+            t.deps_skip.fetch_add(1, Ordering::Relaxed);
```

Comment:
We should do a separate load, add, and store so that we don't have lock instructions.

Reply:
```suggestion
            let deps_skip = t.deps_skip.load(Ordering::Relaxed);
            t.deps_skip.store(deps_skip + 1, Ordering::Relaxed);
```